src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_evp_pkey_rsa.c

```diff
@@ -111,6 +110,82 @@ int32_t CryptoNative_RsaDecrypt(EVP_PKEY* pkey,
     return ret;
 }
 
+int32_t CryptoNative_RsaSignHash(EVP_PKEY* pkey,
+                                 RsaPaddingMode padding,
+                                 const EVP_MD* digest,
+                                 const uint8_t* hash,
+                                 int32_t hashLen,
+                                 uint8_t* destination,
+                                 int32_t destinationLen)
+{
+    assert(pkey != NULL);
+    assert(hash != NULL);
```

Comment:
I think I traced back to the public `RSAOpenSsl.TrySignHash` entry point that _hash_ can be _nullptr_, since `ROS<byte>.IsEmpty` is never checked. Double-check me on this.

Reply:
Hmm.  I agree, I don't see a size check.  So either there's no test with the default span as input, or I'm missing something.  I'll look more into this.

Reply:
Yeah, looks like we're missing a test here.  The assert totally trips.

The good news is, with the test case, I see that simply removing the assert makes things behave as expected:

```
        Interop+Crypto+OpenSslCryptographicException : error:0408E08F:rsa routines:pkey_rsa_sign:invalid digest length
        Stack Trace:
          /home/jbarton/git/bartonjs/runtime/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPkey.Rsa.cs(94,0): at Interop.Crypto.RsaSignHash(SafeEvpPKeyHandle pkey, RSASignaturePaddingMode paddingMode, IntPtr digestAlgorithm, ReadOnlySpan`1 hash, Span`1 destination)
          /home/jbarton/git/bartonjs/runtime/src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs(683,0): at System.Security.Cryptography.RSAImplementation.RSAOpenSsl.TrySignHash(ReadOnlySpan`1 hash, Span`1 destination, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding, Boolean allocateSignature, Int32& bytesWritten, Byte[]& signature)
          /home/jbarton/git/bartonjs/runtime/src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs(639,0): at System.Security.Cryptography.RSAImplementation.RSAOpenSsl.TrySignHash(ReadOnlySpan`1 hash, Span`1 destination, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding, Int32& bytesWritten)
```

(That's the correct exception since (NULL, 0) is not a valid input for RSASSA-*-SHA256... which wants something of length 32.)